code/modules/paperwork/ticketmachine.dm

```diff
@@ -5,234 +5,182 @@
 		user.adjust_fire_stacks(1)
 		user.IgniteMob()
 		return
 
+/obj/machinery/ticket_machine/attackby(obj/item/O, mob/user, params)
+	if(user.a_intent == INTENT_HARM) //so we can hit the machine
+		return ..()
+
+	if(default_deconstruction_screwdriver(user, "ticketmachine_panel", "ticketmachine", O))
+		updateUsrDialog()
+		return TRUE
+
+	if(default_deconstruction_crowbar(O))
+		return TRUE
+
+	if(stat)
+		return TRUE
+
+	if(istype(O, /obj/item/ticket_machine_remote))
+		if (!linked)
+			var/obj/item/ticket_machine_remote/Z = O //typecasting!!
+			Z.connection=src
+			to_chat(user,"<span class='info'>You link the remote to the machine.</span>")
+			linked = TRUE
+			return TRUE
+		else
+			to_chat(user,"<span class='warning'>The ticket machine is already linked to a remote!</span>")
+
+	if(istype(O, /obj/item/ticket_machine_ticket))
+		to_chat(user, "<span class='warning'>You put [O] into the ticket machine's recycling bin.</span>")
+		qdel(O) //KMC put a delay here. I'm not so forgiving. You accidently shred your ticket, you cry. (Also the delay felt really clunky)
+		return TRUE
+
+/obj/machinery/ticket_machine/Destroy()
+	return ..()
+
+//Tickets dispensed from the machine
 /obj/item/ticket_machine_ticket
 	name = "Ticket"
-	desc = "A ticket which shows your place in the Head of Personnel's line. Made from Nanotrasen patented NanoPaper�. Though solid, its form seems to shimmer slightly. Feels (and burns) just like the real thing."
+	desc = "A ticket which shows your place in the queue."
 	icon = 'icons/obj/bureaucracy.dmi'
 	icon_state = "ticket"
-	maptext_x = 7
-	maptext_y = 10
 	w_class = WEIGHT_CLASS_TINY
 	resistance_flags = FLAMMABLE
 	max_integrity = 50
-	var/saved_maptext = null
-	var/mob/living/carbon/owner
 	var/obj/machinery/ticket_machine/source
 	var/ticket_number
 
-/obj/item/ticket_machine_ticket/attack_hand(mob/user)
-	. = ..()
-	maptext = saved_maptext //For some reason, storage code removes all maptext off objs, this stops its number from being wiped off when taken out of storage.
-
-/obj/item/ticket_machine_ticket/attackby(obj/item/P, mob/living/carbon/human/user, params) //Stolen from papercode
+/obj/item/ticket_machine_ticket/update_icon()
+	var/Temp=ticket_number //this stuff is a repeat from the other update_icon
+	var/Digit1 = round(Temp%10)
+	var/Digit2 = round(((Temp-Digit1)*0.1)%10)
+	var/Digit3 = round(((Temp-Digit1-Digit2*10)*0.01)%10)
+	overlays = list() //since tickkets only update once, this shouldn't be super nessesary, but whatevs
+	overlays += image('icons/obj/bureaucracy_overlays.dmi',icon_state = "ticket_first_[Digit1]")
+	overlays += image('icons/obj/bureaucracy_overlays.dmi',icon_state = "ticket_second_[Digit2]")
+	overlays += image('icons/obj/bureaucracy_overlays.dmi',icon_state = "ticket_third_[Digit3]")
+	if(resistance_flags & ON_FIRE)
+		icon_state = "ticket_onfire"
+
+/obj/item/paper/attackby(obj/item/P, mob/living/carbon/human/user, params)
 	..()
 	if(P.is_hot())
```

Comment:
```suggestion
```

Reply:
you misunderstand. This needs to be two separate things